### PR TITLE
Docs: add a more intricate error scenario that might bite other people

### PR DESCRIPTION
**Pull request by janl**

### CODE REVIEW

File: docs/errors.md [hunk 1]
```diff
@@ -221,5 +221,23 @@ var db = new PouchDB('database', { db: require('leveldown-prebuilt') });
...
 ```
 
+{% include anchor.html class="h3" title="Replication with attachments is slow or fails" hash="replicating_attachments_slow" %}
+
+The symptoms for this issues are:
+
+1. Replicating a database that has many attachments from a CouchDB server is either slow or fails randomly.
+2. You get server error message of the nature `No buffer space available`.
+
+Chances are that your server runs inside a virtual machine. The host system, or hypervisor, imposes limits on how much data each virtual machine can use for networking. If you are on a cheap virutal server, it is possible, that the default settings for PouchDB pull-replication (10 parallel batches of 100 documents each) exhaust the narrow limit of your server. Even a single client can cause this.
```
Comment (amb26) [inline docs/errors.md]: typo "virutal"

File: docs/errors.md [hunk 2]
```diff
@@ -221,5 +221,23 @@ var db = new PouchDB('database', { db: require('leveldown-prebuilt') });
...
+
+The solution is to move to a better server, but if that is not an immediate option, a workaround would be reducing the `options.batch_size` and `options.batches_limit` [replication options](http://pouchdb.com/api.html#replication).
+
+To find optimal values, start by setting them both to 1 (meaning that PouchDB should download one document after the other) and increase from there and stop when the symptoms begin again. Note that multiple concurrent clients can still cause an issue, if you get to many. If all your documents have one or more attachments (e.g. a photos database), setting both options to `1` is probably a good idea.
```
Comment (amb26) [inline docs/errors.md]: typo "to many"

File: docs/errors.md (hunk 1 above)
Comment (janl) [inline docs/errors.md]: fixed, thanks!

File: docs/errors.md (hunk 2 above)
Comment (janl) [inline docs/errors.md]: fixed, thanks!

**Comment by nickcolley** [comment]
This looks like a tricky one to come across, thanks a lot :)

Your link didn't hyperlink up so I fixed that on the way in, also wrapped your closing notes in alert boxes hope that's okay.

Also a tiny thing w/ the commit message according to https://github.com/pouchdb/pouchdb/blob/master/CONTRIBUTING.md#git-essentials

https://github.com/pouchdb/pouchdb/commit/39dff41aab408599bb1851f53fff19c423d26a5a?w=1

:dancer:

**Comment by janl** [comment]
Woot thanks! :)

I didn’t include the ticket number in the commit message because the commit happened before the PR (necessarily). I found it weird, but is the desired procedure to amend the commit after the PR is opened to match the number? (happy to do so, if that’s the deal)

**Comment by marten-de-vries** [comment]
@janl That's one possibility, the other is just upping the issue number in the URL until you hit a 404 and using that one.

**Comment by nolanlawson** [comment]
@marten-de-vries that's what I do. I love the little 404 star wars dude anyway. :)

**Pull request closed without merging**